Index: pkg-tigervnc/vncviewer/Viewport.cxx
===================================================================
--- pkg-tigervnc.orig/vncviewer/Viewport.cxx
+++ pkg-tigervnc/vncviewer/Viewport.cxx
@@ -313,7 +313,7 @@ void Viewport::setCursor(int width, int
     }
   }
 
-  if (Fl::belowmouse() == this)
+  if (Fl::belowmouse() == this && cursor)
     window()->cursor(cursor, cursorHotspot.x, cursorHotspot.y);
 }
 
