Description: Add help output for options -display and -geometry."
Author: <unknown>

Index: pkg-tigervnc/unix/vncserver
===================================================================
--- pkg-tigervnc.orig/unix/vncserver
+++ pkg-tigervnc/unix/vncserver
@@ -678,6 +678,7 @@ sub Usage
 	"                 [-geometry <width>x<height>]\n".
 	"                 [-pixelformat rgbNNN|bgrNNN]\n".
 	"                 [-fp <font-path>]\n".
+	"                 [-cc <visual>]\n".
 	"                 [-fg]\n".
 	"                 [-autokill]\n".
 	"                 [-noxstartup]\n".
Index: pkg-tigervnc/vncviewer/vncviewer.cxx
===================================================================
--- pkg-tigervnc.orig/vncviewer/vncviewer.cxx
+++ pkg-tigervnc/vncviewer/vncviewer.cxx
@@ -371,6 +371,11 @@ static void usage(const char *programNam
 #endif
 
   fprintf(stderr,"\n"
+	  "Options:\n\n"
+	  "  -display Xdisplay - Specifies the X display for the viewer window\n"
+	  "  -geometry geometry - Standard X position and sizing specification.\n");
+
+  fprintf(stderr,"\n"
           "Parameters can be turned on with -<param> or off with -<param>=0\n"
           "Parameters which take a value can be specified as "
           "-<param> <value>\n"
