Description: Find fltk shared libraries
Author: <unkonwn>

Index: pkg-tigervnc/CMakeLists.txt
===================================================================
--- pkg-tigervnc.orig/CMakeLists.txt
+++ pkg-tigervnc/CMakeLists.txt
@@ -231,7 +231,8 @@ endif()
 set(FLTK_SKIP_FLUID TRUE)
 set(FLTK_SKIP_OPENGL TRUE)
 set(FLTK_SKIP_FORMS TRUE)
-find_package(FLTK)
+#find_package(FLTK)
+set (FLTK_LIBRARIES "fltk_images;fltk")
 
 if(UNIX AND NOT APPLE)
   # No proper handling for extra X11 libs that FLTK might need...
